Misuse of non-blocking sockets

When calling `setblocking(0)`, one has to expect that operations fail with `WSAEWOULDBLOCK` (10035). The code currently doesn't seem to account for that, making the plugin unusable with remote servers that don't respond immediately.

```
[idarling][INFO] Connecting to idarling:31013...
[idarling][ERROR] [WinError 10035] A non-blocking socket operation could not be completed immediately
Traceback (most recent call last):
  File "C:/Program Files/IDA 7/plugins\idarling\network\network.py", line 114, in connect
    sock.connect((host, port))
BlockingIOError: [WinError 10035] A non-blocking socket operation could not be completed immediately
[idarling][INFO] Disconnected
```

There are two possible solutions:
a) Use `sock.setblocking(1)` in `network.py`.
b) Properly deal with asynchronous operations.

--- a/idarling/network/network.py
+++ b/idarling/network/network.py
@@ -113,8 +113,9 @@ class Network(Module):
         try:
             sock.connect((host, port))
         except OSError as e:
-            self._plugin._logger.exception(e)
-            self._client.disconnect()
+            if e.errno not in (errno.EINPROGRESS, errno.EWOULDBLOCK):
+                self._plugin._logger.exception(e)
+                self._client.disconnect()
 
     def disconnect(self):
         """Disconnect from the current server."""

